ui/Reviews.php

```diff
@@ -139,14 +150,63 @@ public function viewRecentReviews() {
             $('.sortable-table').tablesorter({
                 sortList: [[INITIAL_SORT_COL, 0]],
             });
+
+            function setGenreVisibility(genre, showIt) {
+                let genreClass = 'tr.' + genre;
+                showIt ?  $(genreClass).show() : $(genreClass).hide();
+            }
+
+            let genreMap = {};
+            let reviewCnt = 0;
+            $('.sortable-table > tbody > tr').each(function(e) {
+                reviewCnt++;
+                let genre = $(this).data('genre');
+                if (genreMap[genre] === undefined) {
+                    genreMap[genre] = 0;
+                    $(".review-categories span." + genre).removeClass('zk-hidden');
+                }
+                genreMap[genre]++;
+            });
+            $("span.review-categories").removeClass('zk-hidden');
+            $("#review-count").text(' Found ' + reviewCnt + ' reviews.');
+
+            for (let [genre, count] of Object.entries(genreMap)) {
+                $(`span.${genre} > span`).text(`${genre} (${count})`);
+            }
+
+            let selectedDj = $('#djPicker').children("option:selected").val();
+            let storageKey = 'ReviewCategories-' + selectedDj;
+            let categoryStr = localStorage.getItem(storageKey);
+            let categories = categoryStr ? JSON.parse(categoryStr) : {};
+                
+            $(".categoryPicker input").each(function(e) {
+                let genre  = $(this).val();
+                let isChecked = !(categories[genre] === false);
+                setGenreVisibility(genre, isChecked)
+                $(this).prop('checked', isChecked);
+            });
+
+            $("#djPicker").on('change', function(e) {
+                let selectedDj = $(this).children("option:selected").val();
+                window.location.assign('/?action=viewRecent&dj=' + selectedDj);
```

Comment:
Nice!  Looking good so far.

One thing is this URL.  Needs to be '?action=...' (drop the initial slash).

Can you please fix?  Thanks.

Reply:
done.